src/main/java/org/spongepowered/api/util/Ticks.java

```diff
@@ -0,0 +1,222 @@
+     * this represents will be higher.</p>
+     *
+     * @param time The time
+     * @param temporalUnit The {@link TemporalUnit} of time given in {@code time}.
+     * @return The {@link Ticks} that represents the number of ticks that would
+     *      be expected to be run in an ideal scenario.
+     */
+    static Ticks of(final int time, final TemporalUnit temporalUnit) {
+        return Sponge.getRegistry().getFactoryRegistry().provideFactory(Factory.class).of(time, temporalUnit);
+    }
+
+    /**
+     * Returns a {@link Ticks} that would be executed in the given number of
+     * seconds in an ideal case.
+     *
+     * @param seconds The number of seconds
+     * @return The {@link Ticks} that represents the number of ticks that would
+     *      be expected to be run in an ideal scenario.
+     */
+    static Ticks ofSeconds(final int seconds) {
+        return Ticks.of(seconds, ChronoUnit.SECONDS);
+    }
+
+    /**
+     * Returns a {@link Ticks} that would be executed in the given number of
+     * minutes in an ideal case.
+     *
+     * @param minutes The number of seconds
+     * @return The {@link Ticks} that represents the number of ticks that would
+     *      be expected to be run in an ideal scenario.
+     */
+    static Ticks ofMinutes(final int minutes) {
+        return Ticks.of(minutes, ChronoUnit.MINUTES);
+    }
+
+    /**
+     * Returns a {@link Ticks} that would be executed in the given number of
+     * hours in an ideal case.
+     *
+     * @param hours The number of hours
+     * @return The {@link Ticks} that represents the number of ticks that would
+     *      be expected to be run in an ideal scenario.
+     */
+    static Ticks ofHours(final int hours) {
+        return Ticks.of(hours, ChronoUnit.HOURS);
+    }
+
+    /**
+     * Returns a {@link Ticks} that would be executed in the given number of
+     * days in an ideal case.
+     *
+     * @param days The number of days
+     * @return The {@link Ticks} that represents the number of ticks that would
+     *      be expected to be run in an ideal scenario.
+     */
+    static Ticks ofDays(final int days) {
+        return Ticks.of(days, ChronoUnit.DAYS);
+    }
+
+    /**
+     * Returns a {@link Ticks} that would correspond to the supplied number of
+     * in game days, rounded to the nearest tick.
+     */
+    static Ticks ofMinecraftDays(final double days) {
+        return Sponge.getRegistry().getFactoryRegistry().provideFactory(Factory.class).ofMinecraftDays(days);
+    }
+
+    /**
+     * Returns a {@link Ticks} that would correspond to the supplied number of
+     * in game hours, rounded to the nearest tick.
+     */
+    static Ticks ofMinecraftHours(final double hours) {
+        return Sponge.getRegistry().getFactoryRegistry().provideFactory(Factory.class).ofMinecraftDays(hours / 24.0);
+    }
+
+    /**
+     * Gets the expected {@link Duration} of this {@link Ticks} if the server
```

Comment:
Not sure if this should mention overloading?

Reply:
It makes sense to do so in a way as the time will change if there is any overloading - this is the whole problem of why doing this is necessary in the first place. However, I wrangled with the naming of this method for ages and I admit, I'm still not happy with it. Some iterations I went for were:

* `getMinimumAllottedDuration`
* `getMaximumExpectedDuration`
* `getExpectedDuration` (which is what I've got here)

but none of these are particularly clear because really, the point is that a tick isn't really this time unit but people are using it as one.